website/archiver/decorators.py

```diff
@@ -0,0 +1,25 @@
+import functools
+
+from framework.exceptions import HTTPError
+
+from website.project.decorators import _inject_nodes
+from website.archiver import ARCHIVER_UNCAUGHT_ERROR
+from website.archiver import utils
+
+def fail_archive_on_error(func):
+
+    @functools.wraps(func)
+    def wrapped(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except HTTPError as e:
+            _inject_nodes(kwargs)
```

Comment:
You may want have this decorator imply others if it requires a node to be defined.